include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,277 @@
+        size_t sequence_size_after{0};
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size_before = ranges::size(sequence);
+
+        auto constexpr is_tab{is_char<'\t'>};
+        auto constexpr is_cntrl{is_char<'\n'>};
+
+        while (is_char<'@'>(*ranges::begin(stream_view)))
+            detail::consume(stream_view | view::take_line);
+
+        /* ID */
+        if (is_char<'*'>(*begin(stream_view)))
+            throw parse_error{"The ID field may not be empty for sequence files."};
+
+        if constexpr (!detail::decays_to_ignore_v<id_type>)
+        {
+            if (options.truncate_ids)
+            {
+                std::ranges::copy(stream_view | view::take_until_or_throw(is_blank)
+                                         | view::char_to<value_type_t<id_type>>,
+                             std::back_inserter(id));
+                detail::consume(stream_view | view::take_until_or_throw(is_tab));
+            }
+            else
+            {
+                std::ranges::copy(stream_view | view::take_until_or_throw(is_tab)
+                                         | view::char_to<value_type_t<id_type>>,
+                             std::back_inserter(id));
+            }
+        }
+        else
+        {
+             detail::consume(stream_view | view::take_until_or_throw(is_tab));
+        }
+
+        //Jump over SAM colums 2-9 (FLAG RNAME POS MAPQ CIGAR RNEXT PNEXT TLEN)
+        for (int i = 0; i < 8;i++)
+        {
+            ++begin(stream_view);
+            detail::consume(stream_view | view::take_until_or_throw(is_tab));
+        }
+        ++begin(stream_view);
+
+        auto seq_view{stream_view | view::take_until_or_throw(is_tab)}; // until next tab
+
+         // Sequence
+         if (is_char<'*'>(*begin(stream_view)))
+             throw parse_error{"The Sequence field may not be empty for sequence files."};
+
+         if constexpr (!detail::decays_to_ignore_v<seq_type>)
+         {
+             auto constexpr is_legal_alph = is_in_alphabet<seq_legal_alph_type>;
+             std::ranges::copy(seq_view | view::transform([is_legal_alph] (char const c) // enforce legal alphabet
+                                      {
+                                          if (!is_legal_alph(c))
+                                          {
+                                              throw parse_error{std::string{"Encountered an unexpected letter: "} +
+                                                                is_legal_alph.msg.string() +
+                                                                " evaluated to false on " +
+                                                                detail::make_printable(c)};
+                                          }
+                                          return c;
+                                      })
+                                    | view::char_to<value_type_t<seq_type>>, // convert to actual target alphabet
```

Comment:
this indented weirdly. Make the `|` align and make the lambas block align with the beginning of `view::` above.